src/components/sign-in.js

```diff
@@ -1,59 +1,183 @@
+                {/* Your Website */}
+                Git Repo
+            </Link>{' '}
+            {new Date().getFullYear()}
+            {'.'}
+        </Typography>
+    );
+}
+
+export default function SignInForm() {
+    const classes = useStyles();
+        const formMethods = useForm();
+        const { signIn } = useAuth();
+        const navigate = useNavigate();
+
+        const {
+            control,
+            register,
+            handleSubmit,
+            formState: { errors },
+        } = formMethods;
+
+        const [isLoading, setIsLoading] = useState(false);
+        const [loginError, setLoginError] = useState(null);
+
+
+        const onSubmit = async (data) => {
+            setIsLoading(true);
+            setLoginError(null);
+            try {
+                await signIn(data);
+                setIsLoading(false);
+                navigate("/tasks");
+            } catch (error) {
+                setIsLoading(false);
+                // handle error, e.g. show a message to the user
+                setLoginError("Invalid username or password.");
+            }
+        };
+
+    // Error Message Div:
+    const errorDiv =loginError
+        ? <Alert severity="error">{loginError}</Alert>
+        : '';
+
+    const [showPassword, setShowPassword] = useState(false);
+
+    const handleClickShowPassword = () => setShowPassword((show) => !show);
+
+    const handleMouseDownPassword = (event) => {
+        event.preventDefault();
+    };
+
+
+    return (
+        <Container component="main" maxWidth="xs">
+            <CssBaseline/>
+            <div className={classes.paper}>
+                <Avatar className={classes.avatar}>
+                    <LockOutlinedIcon/>
+                </Avatar>
+                <Typography component="h1" variant="h5">
+                    Sign In
+                </Typography>
+                <form className={classes.form} onSubmit={handleSubmit(onSubmit)}>
+                    <Grid container spacing={2}>
+                        <Grid item xs={12}>{errorDiv}</Grid>
+                        <Grid item xs={12}>
+                            <InputField
+                                {...register("username")}
+                                name="username"
+                                control={control}
+                                defaultValue=""
+                                rules={{
+                                    required: 'Username is required.',
+                                }}
+                                autoComplete="email"
+                                label="Username"
+                                error={!!errors.email}
+                                helperText={errors.email?.message}
+                            />
+                        </Grid>
+                        <Grid item xs={12}>
+                            <TextField
+                                {...register("password")}
```

Comment:
We're using TextField here and InputField above. Lets change this to InputField

Reply:
The Inputfield doesn't work for the InputProps to show the eye icon for the password field, i tried different ways, no success, so had to manually use directly here as text field.